apps/nsq_to_file/file_logger.go

```diff
@@ -164,6 +164,19 @@ func (f *FileLogger) Close() {
 			f.gzipWriter.Close()
 		}
 		f.out.Close()
+
+		// Move file from work dir to output dir if necessary
+		if *workDir != *outputDir {
+			src := f.out.Name()
+			dst := strings.Replace(src, *workDir, *outputDir, 1)
```

Comment:
nit, if your working directory is `a` this replace won't be so pretty. Can we use `filepath.Split` on `f.out.Name` here?

Reply:
Ah, very good point!

That was actually the first approach I took here, but I found that it breaks down for the case where `-filename-format` itself contains a directory, like

```
-filename-format="<TOPIC>/<TOPIC>.<DATETIME>+0000.<HOST><REV>.log"
```

In that case, a naive use of `filepath.Split()` drops the topic directory from the final output destination path.

Would an approach along these lines be good enough?

```go
dst := filepath.Join(*outputDir, strings.TrimPrefix(src, *workDir))
```

Reply:
(Updated w/ that new approach)

Reply:
👍 good catch that filename-format can have directories; it's not "just" a filename. `TrimPrefix` sounds like it will work.